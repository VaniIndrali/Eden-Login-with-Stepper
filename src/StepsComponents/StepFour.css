.subText {
	margin-top: 13px;
}

.mainText {
	margin-top: 1.75rem;
}

.iconClass {
	margin-top: 3.125rem;
	text-align: center;
}
