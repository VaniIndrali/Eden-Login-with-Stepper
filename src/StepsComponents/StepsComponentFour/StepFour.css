.displaySubText {
	margin-top: 0.813rem;
	font-size: 0.9rem;
	margin-bottom: 1.875rem;
}

.displayText {
	margin-top: 1.75rem;
	font-weight: 600;
}

.iconClass {
	margin-top: 3.125rem;
	text-align: center;
}

.circle_tick {
	height: 3.5rem;
	width: 4.1rem;
}
