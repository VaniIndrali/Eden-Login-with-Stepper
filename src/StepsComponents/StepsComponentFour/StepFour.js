import { React } from "react";
import "./StepFour.css";

export const StepFour = (props) => {
	const { displayName } = props;

	const handleSubmit = (e) => {
		e.preventDefault();
	};

	return (
		<div id="loginFour">
			<div className="container">
				<div className="iconClass">
					<img
						src="https://i.imgur.com/gwxWVOi.jpg"
						className="circle_tick"
						alt="circletick"
					/>
				</div>
				<h2 className="text-center displayText">
					Congratulations, {displayName}!
				</h2>
				<h6 className="text-center displaySubText">
					You have completed onboarding, you can start using the Eden!
				</h6>
				<div
					id="login-row"
					className="row justify-content-center align-items-center"
				>
					<div
						id="login-column"
						className="col-md-11"
					>
						<div
							id="login-box"
							className="col-md-12"
						>
							<form
								id="login-form"
								className="form"
								onSubmit={(e) => handleSubmit(e)}
							>
								<div className="form-group">
									<input
										type="submit"
										name="Launch Eden"
										className="button"
										value="Launch Eden"
										id="buttonEden"
									/>
								</div>
							</form>
						</div>
					</div>
				</div>
			</div>
		</div>
	);
};

export default StepFour;
