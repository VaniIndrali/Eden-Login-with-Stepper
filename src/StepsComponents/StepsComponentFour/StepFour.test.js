import { render, screen } from "@testing-library/react";
import StepFour from "./StepFour";

test("renders StepFour", () => {
	const props = {
		displayName: "Joseph",
		onSubmit: jest.fn(),
	};

	const component = render(<StepFour {...props} />);

	expect(
		screen.getByRole("heading", {
			name: "Congratulations, Joseph!",
		})
	).toBeTruthy();

	expect(
		screen.getByRole("heading", {
			name: "You have completed onboarding, you can start using the Eden!",
		})
	).toBeTruthy();

	expect(
		screen.getByRole("button", {
			name: "Launch Eden",
		})
	).toBeTruthy();

	expect(component).toMatchSnapshot();
});
