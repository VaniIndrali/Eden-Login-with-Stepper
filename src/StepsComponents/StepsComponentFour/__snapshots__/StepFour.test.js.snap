// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders StepFour 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        id="loginFour"
      >
        <div
          class="container"
        >
          <div
            class="iconClass"
          >
            <img
              alt="circletick"
              class="circle_tick"
              src="https://i.imgur.com/gwxWVOi.jpg"
            />
          </div>
          <h2
            class="text-center displayText"
          >
            Congratulations, 
            Joseph
            !
          </h2>
          <h6
            class="text-center displaySubText"
          >
            You have completed onboarding, you can start using the Eden!
          </h6>
          <div
            class="row justify-content-center align-items-center"
            id="login-row"
          >
            <div
              class="col-md-11"
              id="login-column"
            >
              <div
                class="col-md-12"
                id="login-box"
              >
                <form
                  class="form"
                  id="login-form"
                >
                  <div
                    class="form-group"
                  >
                    <input
                      class="button"
                      id="buttonEden"
                      name="Launch Eden"
                      type="submit"
                      value="Launch Eden"
                    />
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      id="loginFour"
    >
      <div
        class="container"
      >
        <div
          class="iconClass"
        >
          <img
            alt="circletick"
            class="circle_tick"
            src="https://i.imgur.com/gwxWVOi.jpg"
          />
        </div>
        <h2
          class="text-center displayText"
        >
          Congratulations, 
          Joseph
          !
        </h2>
        <h6
          class="text-center displaySubText"
        >
          You have completed onboarding, you can start using the Eden!
        </h6>
        <div
          class="row justify-content-center align-items-center"
          id="login-row"
        >
          <div
            class="col-md-11"
            id="login-column"
          >
            <div
              class="col-md-12"
              id="login-box"
            >
              <form
                class="form"
                id="login-form"
              >
                <div
                  class="form-group"
                >
                  <input
                    class="button"
                    id="buttonEden"
                    name="Launch Eden"
                    type="submit"
                    value="Launch Eden"
                  />
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
